index.js

```diff
@@ -176,8 +176,11 @@ Funnel.prototype.build = function() {
   if (this.shouldLinkRoots()) {
     linkedRoots = true;
     if (fs.existsSync(inputPath)) {
-      rimraf.sync(this.outputPath);
-      this._copy(inputPath, this.destPath);
+      // Only re-symlink when either this.destPath doesn't exist or is not a symlink
+      if (!fs.existsSync(this.destPath) || !fs.lstatSync(this.destPath).isSymbolicLink()) {
```

Comment:
@huiyuan and me just discussed this, this can still be improved. We only need to stat the `input`, if it has changed (was removed, or created) since the last time, we update the output accordingly. Our output can only change if we changed it, so no need checking if that is the case.